src/compile/facet.ts

```diff
@@ -121,16 +121,16 @@ export class FacetModel extends Model {
     }, {});
   }
 
-  private initAxis(facet: Facet, config: Config, child: Model): Dict<Axis> {
+  private initAxis(facet: Facet, child: Model): Dict<Axis> {
     const model = this;
     return [ROW, COLUMN].reduce(function(_axis, channel) {
       if (facet[channel]) {
         const axisSpec = facet[channel].axis;
         if (axisSpec !== false) {
-          const modelAxis = _axis[channel] = extend({},
-            config.facet.axis,
-            axisSpec === true ? {} : axisSpec || {}
-          );
+          const modelAxis = _axis[channel] = {
+            ...defaultFacetAxisConfig,
```

Comment:
This is still wrong.  Just like unit spec (which is [still wrong too](https://github.com/vega/vega-lite/pull/1944/files#r101435512)), you shouldn't copy the whole config.